packages/compiler-core/src/codegen.ts

```diff
@@ -760,6 +763,20 @@ function genExpression(node: SimpleExpressionNode, context: CodegenContext) {
 
 function genInterpolation(node: InterpolationNode, context: CodegenContext) {
   const { push, helper, pure } = context
+
+  if (
+    node.content.type === NodeTypes.SIMPLE_EXPRESSION &&
+    node.content.constType === ConstantTypes.CAN_STRINGIFY
+  ) {
+    if (node.content.content) {
+      push(JSON.stringify(toDisplayString(evaluateConstant(node.content))))
+    } else {
+      push(`""`)
+    }
```

Comment:
We can't call `evaluateConstant` when `node.content` is `undefined`, so we just emit an empty string (which is the behaviour of `toDisplayString(undefined)"